src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,147 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
 
 export const PeoplePage = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isError, setIsError] = useState(false);
+
+  const [searchParams] = useSearchParams();
+  const currentSort = searchParams.get('sort') || '';
+  const currentOrder = searchParams.get('order') || '';
+  const currentQuery = searchParams.get('query') || '';
+  const currentSex = searchParams.get('sex') || '';
+  const currentCenturies = searchParams.getAll('centuries') || [];
+
+  const getFilteredPeople = (peopleToFitler: Person[]) => {
+    let filteredPeople = peopleToFitler;
+
+    if (currentQuery) {
+      const normalizedQuery = currentQuery.toLowerCase();
+
+      filteredPeople = filteredPeople.filter((person) => {
+        const { name, fatherName, motherName } = person;
+        const normalizedName = name.toLowerCase();
+        const normalizedFatherName = fatherName?.toLowerCase();
+        const normalizedMotherName = motherName?.toLowerCase();
+
+        return (
+          normalizedName.includes(normalizedQuery)
+          || normalizedFatherName?.includes(normalizedQuery)
+          || normalizedMotherName?.includes(normalizedQuery)
+        );
+      });
+    }
+
+    if (currentSex) {
+      filteredPeople = filteredPeople
+        .filter((person) => person.sex === currentSex);
+    }
+
+    if (currentCenturies.length) {
+      const normalizedCenturies = currentCenturies
+        .map((century) => +century * 100);
+
+      filteredPeople = filteredPeople.filter((person) => {
+        const { born } = person;
+
+        return normalizedCenturies.some((century) => {
+          return born <= century && born >= century - 100;
+        });
+      });
+    }
+
+    switch (currentSort) {
+      case 'name':
+        filteredPeople = filteredPeople.sort((a, b) => {
+          return a.name.localeCompare(b.name);
+        });
+        break;
+
+      case 'sex':
+        filteredPeople = filteredPeople.sort((a, b) => {
+          return a.sex.localeCompare(b.sex);
+        });
+        break;
```

Comment:
```suggestion
      case 'name':
      case 'sex':
        filteredPeople = filteredPeople.sort((a, b) => {
          return a[currentSort].localeCompare(b[currentSort]);
        });
        break;
```
we can group some cases